File: backend/src/main/java/org/rent/room/be/service/RoleService.java
package org.rent.room.be.service;

import org.rent.room.be.dto.request.role.CreateRoleRequest;
import org.rent.room.be.dto.request.role.UpdateRoleRequest;
import org.rent.room.be.dto.response.RoleResponse;

import java.util.List;

public interface RoleService {
    List<RoleResponse> getAllRoles();

    RoleResponse createRole(CreateRoleRequest role);

    RoleResponse updateRole(Long id, UpdateRoleRequest role);

    void updateRoleStatus(Long id, boolean active);
}

File: frontend/src/services/roleService.ts
import api from "../config/axios";
import type { ApiResponse } from "./usersService";

export interface RoleResponse {
  roleId: number;
  roleName: string;
  description: string;
  active: boolean;
}

export interface CreateRoleRequest {
  roleName: string;
  description?: string;
  active?: boolean;
}

export interface UpdateRoleRequest {
  roleName: string;
  description?: string;
}

export const roleService = {
  getAllRoles: () => {
    return api.get<any, ApiResponse<RoleResponse[]>>("/roles");
  },

  createRole: (data: CreateRoleRequest) => {
    return api.post<any, ApiResponse<RoleResponse>>("/roles", data);
  },

  updateRole: (id: number, data: UpdateRoleRequest) => {
    return api.put<any, ApiResponse<RoleResponse>>(`/roles/${id}`, data);
  },

  updateStatus: (id: number, active: boolean) => {
    return api.patch<any, ApiResponse<void>>(
      `/roles/${id}/status?active=${active}`,
    );
  },
};
